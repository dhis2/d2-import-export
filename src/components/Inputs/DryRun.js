import i18n from '@dhis2/d2-i18n'
import React from 'react'
import { RadioGroup } from '../form/RadioGroup'

export const OPTION_YES = { value: 'true', label: i18n.t('Yes') }
export const OPTION_NO = { value: 'false', label: i18n.t('No') }
export const DRY_RUN_KEY = 'dryRun'
export const DRY_RUN_DEFAULT_VALUE = OPTION_NO.value

export const DryRun = () => (
    <RadioGroup
        name={DRY_RUN_KEY}
        label={i18n.t('Dry run')}
        options={[OPTION_YES, OPTION_NO]}
    />
)
