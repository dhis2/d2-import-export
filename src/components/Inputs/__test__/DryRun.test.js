import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { DryRun } from '../DryRun'

jest.mock('react-final-form', () => ({
    useField: jest.fn(() => ({
        input: {
            name: 'Name',
            value: '',
            onChange: jest.fn(),
        },
        meta: {
            touched: false,
            error: '',
        },
    })),
}))

describe('Input component - DryRun', () => {
    it('should render correctly', () => {
        const file = shallow(<DryRun />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
