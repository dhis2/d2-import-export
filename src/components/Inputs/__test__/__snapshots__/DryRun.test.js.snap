// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - DryRun should render correctly 1`] = `
<RadioGroup
  label="Dry run"
  name="dryRun"
  options={
    Array [
      Object {
        "label": "Yes",
        "value": "true",
      },
      Object {
        "label": "No",
        "value": "false",
      },
    ]
  }
/>
`;
